diff --git a/node_modules/hyperdrive/index.js b/node_modules/hyperdrive/index.js
index be92c2b..446a581 100644
--- a/node_modules/hyperdrive/index.js
+++ b/node_modules/hyperdrive/index.js
@@ -178,11 +178,11 @@ module.exports = class Hyperdrive extends EventEmitter {
     return this.blobs
   }
 
-  async get (name) {
-    const node = await this.entry(name)
+  async get (name, opts) {
+    const node = await this.entry(name, opts)
     if (!node?.value.blob) return null
     await this.getBlobs()
-    return this.blobs.get(node.value.blob)
+    return this.blobs.get(node.value.blob, opts)
   }
 
   async put (name, buf, { executable = false, metadata = null } = {}) {
@@ -201,9 +201,9 @@ module.exports = class Hyperdrive extends EventEmitter {
     return this.files.put(normalizePath(name), { executable: false, linkname: dst, blob: null, metadata })
   }
 
-  entry (name) {
+  entry (name, opts) {
     return typeof name === 'string'
-      ? this.files.get(normalizePath(name))
+      ? this.files.get(normalizePath(name), opts)
       : Promise.resolve(name)
   }
 
