diff --git a/node_modules/hyperblobs/lib/streams.js b/node_modules/hyperblobs/lib/streams.js
index 41658ba..d6d72ef 100644
--- a/node_modules/hyperblobs/lib/streams.js
+++ b/node_modules/hyperblobs/lib/streams.js
@@ -101,6 +101,9 @@ class BlobReadStream extends Readable {
     else if (opts.end !== undefined) this._end = id.byteOffset + opts.end + 1
     else this._end = id.byteOffset + id.byteLength
 
+    this.wait = opts.wait !== false
+    this.timeout = opts.timeout || 0
+
     this._index = 0
     this._relativeOffset = 0
     this._bytesRead = 0
@@ -143,7 +146,7 @@ class BlobReadStream extends Readable {
       this._lastPrefetch = this.core.download(prefetch)
     }
 
-    this.core.get(this._index).then(block => {
+    this.core.get(this._index, { wait: this.wait, timeout: this.timeout }).then(block => {
       if (!block) return cb(new Error('Block not available'))
 
       const remainder = this._end - this._pos
